[ROCm][CI] Enable pre- and post-merge `default` and `distributed` config ROCm CI jobs on trunk.yml

jithunnair-amd opened the pull request:
* To enable developers to triage PR-based ROCm CI failures on trunk.yml with the runs on main commits, it's helpful to have trunk.yml jobs running on both PRs and main commits. 
* Remove main and release branch triggers for `rocm-mi355.yml` since trunk.yml will provide that coverage for `default` and `distributed` config now
* Remove most cron schedules for `periodic-rocm-mi355.yml` since trunk.yml will provide that coverage for `distributed` config now.

cc @jeffdaily @sunway513 @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang

jithunnair-amd committed:
Enable ROCm jobs in trunk.yml

jithunnair-amd committed:
Remove push/cron triggers

malfet commented on .github/workflows/trunk.yml:
```diff
@@ -295,10 +295,6 @@ jobs:
-    #NS: Trunk jobs should run on both PR if ciflow/trunk is added an on trunk
-    # But https://github.com/pytorch/pytorch/pull/165674 always skips them on trunk
-    if: 0
-    # if: ${{ startsWith(github.event.ref, 'refs/tags/ciflow/trunk') || (needs.job-filter.outputs.jobs != '' && contains(needs.job-filter.outputs.jobs, ' linux-jammy-rocm-py3.10 ')) }}
```
Probably LGTM, but are you sure capacity wouldn't be a problem?
See 
<img width="855" height="378" alt="image" src="https://github.com/user-attachments/assets/24334881-d514-44e4-a274-aa3683607a7e" />

jithunnair-amd replied:
We should only be looking at the `linux.rocm.gpu.gfx950.*` labels (used by trunk.yml) to assess the MI350 capacity situation:
<img width="464" height="212" alt="{4622375B-8AF8-452D-9ED6-9EE46EA41005}" src="https://github.com/user-attachments/assets/f26c7261-8642-45a3-aca0-a311e6ab1615" />

Here's a more detailed chart showing how the MI350 runners have been doing wrt queueing:
https://hud.pytorch.org/queue_time_analysis?dateRange=7&startDate=2026-01-26T18%3A07%3A29.907Z&endDate=2026-02-02T18%3A07%3A29.907Z&granularity=hour&chartType=bar&repos=pytorch%2Fpytorch&category=machine_type&machineTypes=linux.rocm.gpu.gfx950.1&machineTypes=linux.rocm.gpu.gfx950.4&items=linux.rocm.gpu.gfx950.1&items=linux.rocm.gpu.gfx950.4

Based on our cluster utilization dashboards, <15% of our MI350 cluster is being utilized right now by the current level of ROCm CI on PyTorch. So we feel confident that we can absorb the extra workload from trunk.yml that will be enabled by this PR.

malfet reviewed (changes requested):
Not sure what is the intend of inductor-rocm jobs dispatch changes, but it sounds contradictory to PR title

jithunnair-amd commented on .github/workflows/rocm-mi355.yml:
```diff
@@ -2,9 +2,6 @@ name: rocm-mi355
   push:
-    branches:
-      - main
-      - release/*
```
I don't think so. Not having any `branches` keyword in there means that only pushes to tags matching `ciflow/rocm-mi355` should trigger this workflow. cc @atalman to confirm if this matches his understanding.

jithunnair-amd replied:
I'm checking if the
```
  schedule:
    - cron: 29 8 * * *  # about 1:29am PDT
```
clause will run on every branch though...

jithunnair-amd replied:
Looks like the `cron` trigger will only run on the default branch: https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows
`Scheduled workflows run on the latest commit on the default branch.`, so even the `cron` trigger is not a concern.

This is validated by the fact that `nightly.yml` scheduled runs are only triggered on `main` branch: https://github.com/pytorch/pytorch/actions/workflows/nightly.yml?query=event%3Aschedule

malfet reviewed (approved):
LGTM, but would be curious to figure out how one can estimate the capacity needed to run trunk tests

jithunnair-amd commented:
> LGTM, but would be curious to figure out how one can estimate the capacity needed to run trunk tests

We used the [Flambeau ](https://hud.pytorch.org/flambeau) tool to inform our calculations.

jithunnair-amd committed:
inductor-rocm-mi355.yml should retain the push-to-branch triggers because trunk.yml will *not* run inductor config jobs

jithunnair-amd commented:
@pytorchbot merge -f "Addressed all review comments. trunk.yml triggered via this PR (as expected) and passed in https://hud.pytorch.org/pytorch/pytorch/pull/170294?sha=912113ab216a5d369436b0b1987c461512c9d8ec#trunk"
